services/state/contract/storage_slot.proto

```diff
@@ -0,0 +1,70 @@
+syntax = "proto3";
+
+package proto;
+
+/*-
+ * ‌
+ * Hedera Network Services Protobuf
+ * ​
+ * Copyright (C) 2018 - 2023 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import "state/common.proto";
+
+option java_package = "com.hederahashgraph.api.proto.java";
+// <<<pbj.java_package = "com.hedera.hapi.node.state.contract">>> This comment is special code for setting PBJ Compiler java package
+option java_multiple_files = true;
+
+/**
+ * The key of a storage slot. A slot is scoped to a specific contract number.
+ * 
+ * For each contract, its EVM storage is a mapping of 256-bit keys (or "words") to 256-bit values. 
+ */
+message SlotKey {
```

Comment:
I'm curious where these two messages are used.  Two questions come up:  Why isn't there just one message - what is the use case where you need keys without their corresponding slot or vice-versa?  And second, what is the use case where the contract number is associated with a single key, rather than having a single contract's collection of K/V pairs?  IOW, why is contract#  in the same message as slot#?

Reply:
We want to be able to use a single `VirtualMap` for the storage of all contracts in state; so the key has to be contract-scoped.